bindings/java/src/org/sleuthkit/datamodel/Account.java

```diff
@@ -18,23 +18,182 @@
+		}
+		
+		/**
+		 * Constructs an Account type.
+		 * @param typeName    The type name.
+		 * @param displayName The display name for the type.
+		 */
+		public Type(String typeName, String displayName) {
+			this(0, typeName, displayName );
+		}
+		
+		/**
+		 * Gets the type name
+		 *
+		 * @return The type name.
+		 */
+		public String getTypeName() {
+			return this.typeName;
+		}
+		
 		public String getDisplayName() {
 			return displayName;
 		}
+		
+		/**
+		 * Gets the type id of this account type.
+		 *
+		 * @return The type id.
+		 */
+		public int getTypeID() {
+			return this.typeID;
+		}
 
-		private final String displayName;
+		
 
-		private Type(String displayName) {
-			this.displayName = displayName;
+		@Override
+		public boolean equals(Object that) {
+			if (this == that) {
+				return true;
+			} else if (!(that instanceof Account.Type)) {
+				return false;
+			} else {
+				return ((Account.Type) that).sameType(this);
+			}
+		}
+		
+		@Override
+		public String toString() {
+			return "(typeID= " + this.typeID
+					+ ", displayName=" + this.displayName
+					+ ", typeName=" + this.typeName + ")";
+					
+		}
+		
+		/**
+		 * Determines if this account type object is equivalent to another
+		 * account type object.
+		 *
+		 * @param that the other type
+		 *
+		 * @return true if it is the same type
+		 */
+		private boolean sameType(Account.Type that) {
+			return this.typeName.equals(that.getTypeName())
+					&& this.displayName.equals(that.getDisplayName())
+					&& this.typeID == that.getTypeID();
 		}
 	}
 
+	public Account(SleuthkitCase sleuthkitCase, long artifactId) throws TskCoreException {
+		
+		this.sleuthkitCase = sleuthkitCase;
+		this.artifactId = artifactId;
+		
+		getArtifact();
+		
+		this.accountType =  sleuthkitCase.getAccountType(artifact.getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_ACCOUNT_TYPE)).getValueString());
+		this.accountID = artifact.getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_ID)).getValueString();
+	
+	}
+	
+	public Account(SleuthkitCase sleuthkitCase, long artifactId, Account.Type accountType, String accountID) throws TskCoreException{
+		this.sleuthkitCase = sleuthkitCase;
+		this.artifactId = artifactId;
+		
+		getArtifact();
+		
+		this.accountType = accountType;
+		this.accountID = accountID;
+	}
+	
+	public String getAccountID() {
+		return this.accountID;
+	}
+	
+	public Account.Type getAccountType() {
+		return this.accountType;
+	}
+	
+	public BlackboardAttribute  getAttribute(BlackboardAttribute.ATTRIBUTE_TYPE attrType) throws TskCoreException {
+		return getArtifact().getAttribute(new BlackboardAttribute.Type(attrType));
+	}
+	
+	public void addAttribute(BlackboardAttribute bbatr) throws TskCoreException {
+		getArtifact().addAttribute(bbatr);
+	}
+	
+	public long getArtifactId() {
+		return this.artifactId;
+	}
+	
+	private BlackboardArtifact getArtifact() throws TskCoreException {
+		if (null == this.artifact) {
```

Comment:
What is the point of this "lazy loading" if this method is called in the constructors?